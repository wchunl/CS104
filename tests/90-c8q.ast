ROOT "" (0.0.0)
|   TOK_CALL "(" (4.4.28)
|   |   TOK_IDENT "Q" (4.4.27) variable
|   |   TOK_INTCON "0" (4.4.29) int const
|   |   |   TOK_INTCON "0" (4.4.31) int const
|   |   |   TOK_INTCON "0" (4.4.33) int const
