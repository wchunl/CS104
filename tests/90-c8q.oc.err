xؕ:1.6: syntax error, unexpected TOK_IDENT
xؕ:1.19: syntax error, unexpected '{'
xؕ:2.2: invalid source character (|)
xؕ:2.15: syntax error, unexpected ';'
xؕ:2.19: invalid source character (?)
xؕ:2.20: invalid source character (!)
xؕ:2.21: syntax error, unexpected TOK_IDENT
xؕ:2.33: invalid source character (:)
xؕ:3.14: invalid source character (&)
xؕ:3.16: invalid source character (|)
xؕ:3.18: invalid source character (&)
xؕ:3.20: invalid source character (|)
xؕ:3.21: invalid source character (|)
xؕ:3.29: invalid source character (|)
xؕ:3.33: invalid source character (|)
xؕ:4.18: syntax error, unexpected '{'
xؕ:4.36: syntax error, unexpected '}'
