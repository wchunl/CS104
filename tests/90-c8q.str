string_set[   0]:    4706206694200543916 0xe9fb88->","
string_set[   1]:    2297668033614959926 0xea0618->"0"
                    6637313742931709005 0xea0488->"*"
string_set[   3]:   11868120863286193964 0xea0578->"main"
                    1451333427496801020 0xea0398->"write"
string_set[   4]:   10838281452030117757 0xe9fce8->"b"
                      21606623943142546 0xe9fd88->"{"
string_set[   6]:   13686721134187162920 0xea0528->"}"
                    6407148625343136198 0xe9fc48->"for"
string_set[   8]:    1370730858159036685 0xea03b8->"+"
string_set[   9]:   14526079891305879776 0xea02f8->">"
                    1547029504053110495 0xe9b578->"O"
string_set[  12]:   15816506932648033945 0xe9fa48->"("
string_set[  15]:     747352984659061831 0xea00a8->"q"
string_set[  17]:   17899137375351314596 0xe9ffb8->"8"
string_set[  21]:   11540658564628934174 0xe9fae8->"l"
                   13581035385482130682 0xe9b218->"["
string_set[  22]:    8243136235647551691 0xea03d8->"56"
string_set[  23]:   12868445110721718657 0xe9fe48->")"
                   12051031859636400188 0xe9f958->"]"
string_set[  24]:     905566266843721762 0xea0348->"7"
                    5343826627144871577 0xe9f9f8->"Q"
string_set[  26]:     342363981814211588 0xe9f648->"char"
string_set[  27]:    6623666755989985924 0xe9f8d8->"9"
string_set[  29]:    2594088158685378223 0xea05c8->"10"
string_set[  30]:    8832308542017959627 0xe9fbd8->"22"
string_set[  31]:    6142509188972423790 0xe9b558->""
                   13201088988302871718 0xe9fe68->"o"
string_set[  32]:   11597697714117577063 0xe9ff68->"="
string_set[  34]:    7372855250431704371 0xe9fc28->"-"
string_set[  38]:   12597418120160561823 0xe9fe88->"int"
                   13458138947861825371 0xea0008->"p"
string_set[  42]:    5344662657899890615 0xe9f9a8->";"
string_set[  43]:    4551451650890805270 0xea04d8->"2"
string_set[  45]:    1396094175042499165 0xea00f8->"<"
                   10159970873491820195 0xea0058->"1"
                   14494284460613645429 0xe9fda8->"d"
load_factor = 0.787
bucket_count = 47
max_bucket_size = 3
