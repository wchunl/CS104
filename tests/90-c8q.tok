#  1 "90-c8q.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "90-c8q.oc"
   4 1.00001 282 TOK_IDENT     char
   4 1.00006 282 TOK_IDENT     O
   4 1.00007  91 '['           [
   4 1.00008 283 TOK_INTCON    9
   4 1.00009  93 ']'           ]
   4 1.00010  59 ';'           ;
   4 1.00011 282 TOK_IDENT     Q
   4 1.00012  40 '('           (
   4 1.00013 282 TOK_IDENT     l
   4 1.00014  44 ','           ,
   4 1.00015 282 TOK_IDENT     b
   4 1.00016  44 ','           ,
   4 1.00017 282 TOK_IDENT     d
   4 1.00018  41 ')'           )
   4 1.00019 123 '{'           {
   4 1.00020 262 TOK_INT       int
   4 1.00024 282 TOK_IDENT     o
   4 1.00025  61 '='           =
   4 1.00026 283 TOK_INTCON    8
   4 1.00027  44 ','           ,
   4 1.00028 282 TOK_IDENT     p
   4 1.00029  61 '='           =
   4 1.00030 283 TOK_INTCON    1
   4 1.00031  44 ','           ,
   4 1.00032 282 TOK_IDENT     q
   4 1.00033  61 '='           =
   4 1.00034 283 TOK_INTCON    1
   4 1.00035  60 '<'           <
   4 1.00036  60 '<'           <
   4 2.00001 282 TOK_IDENT     l
   4 2.00003 283 TOK_INTCON    1
   4 2.00004  60 '<'           <
   4 2.00005  60 '<'           <
   4 2.00006 283 TOK_INTCON    22
   4 2.00008  45 '-'           -
   4 2.00009 282 TOK_IDENT     l
   4 2.00010  59 ';'           ;
   4 2.00011 282 TOK_IDENT     for
   4 2.00014  40 '('           (
   4 2.00015  59 ';'           ;
   4 2.00016 282 TOK_IDENT     l
   4 2.00017  62 '>'           >
   4 2.00018 283 TOK_INTCON    7
   4 2.00021 282 TOK_IDENT     write
   4 2.00026  40 '('           (
   4 2.00027 283 TOK_INTCON    1
   4 2.00028  44 ','           ,
   4 2.00029 282 TOK_IDENT     O
   4 2.00030  44 ','           ,
   4 2.00031 283 TOK_INTCON    9
   4 2.00032  41 ')'           )
   4 2.00034 282 TOK_IDENT     o
   4 2.00035  45 '-'           -
   4 2.00036  45 '-'           -
   4 3.00001  59 ';'           ;
   4 3.00002  41 ')'           )
   4 3.00003 282 TOK_IDENT     O
   4 3.00004  91 '['           [
   4 3.00005 282 TOK_IDENT     l
   4 3.00006  93 ']'           ]
   4 3.00007  61 '='           =
   4 3.00008 283 TOK_INTCON    56
   4 3.00010  45 '-'           -
   4 3.00011 282 TOK_IDENT     o
   4 3.00012  44 ','           ,
   4 3.00013 282 TOK_IDENT     b
   4 3.00015 282 TOK_IDENT     p
   4 3.00017 282 TOK_IDENT     d
   4 3.00019 282 TOK_IDENT     q
   4 3.00022 282 TOK_IDENT     Q
   4 3.00023  40 '('           (
   4 3.00024 282 TOK_IDENT     l
   4 3.00025  43 '+'           +
   4 3.00026 283 TOK_INTCON    1
   4 3.00027  44 ','           ,
   4 3.00028 282 TOK_IDENT     b
   4 3.00030 282 TOK_IDENT     p
   4 3.00031  44 ','           ,
   4 3.00032 282 TOK_IDENT     d
   4 3.00034 282 TOK_IDENT     q
   4 3.00035  41 ')'           )
   4 3.00036  44 ','           ,
   4 4.00001 282 TOK_IDENT     p
   4 4.00002  42 '*'           *
   4 4.00003  61 '='           =
   4 4.00004 283 TOK_INTCON    2
   4 4.00005  44 ','           ,
   4 4.00006 282 TOK_IDENT     q
   4 4.00007  42 '*'           *
   4 4.00008  61 '='           =
   4 4.00009 283 TOK_INTCON    2
   4 4.00010  59 ';'           ;
   4 4.00011 125 '}'           }
   4 4.00012 282 TOK_IDENT     main
   4 4.00016  40 '('           (
   4 4.00017  41 ')'           )
   4 4.00018 123 '{'           {
   4 4.00019 282 TOK_IDENT     O
   4 4.00020  91 '['           [
   4 4.00021 283 TOK_INTCON    8
   4 4.00022  93 ']'           ]
   4 4.00023  61 '='           =
   4 4.00024 283 TOK_INTCON    10
   4 4.00026  59 ';'           ;
   4 4.00027 282 TOK_IDENT     Q
   4 4.00028  40 '('           (
   4 4.00029 283 TOK_INTCON    0
   4 4.00030  44 ','           ,
   4 4.00031 283 TOK_INTCON    0
   4 4.00032  44 ','           ,
   4 4.00033 283 TOK_INTCON    0
   4 4.00034  41 ')'           )
   4 4.00035  59 ';'           ;
   4 4.00036 125 '}'           }
