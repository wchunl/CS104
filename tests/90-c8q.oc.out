; 0.  1: # 1 "90-c8q.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "90-c8q.oc"
; 4.  1: char O[9];Q(l,b,d){int o=8,p=1,q=1<<
; 4.  2: l|1<<22-l;for(;l>7?!write(1,O,9):o--
; 4.  3: ;)O[l]=56-o,b&p|d&q||Q(l+1,b|p,d|q),
; 4.  4: p*=2,q*=2;}main(){O[8]=10;Q(0,0,0);}
hello